6.graph/max-area-of-island/step1.cpp

```diff
@@ -0,0 +1,56 @@
+#include <queue>
+#include <vector>
+
+class Solution {
+public:
+  int maxAreaOfIsland(std::vector<std::vector<int>>& grid) {
+    int height = grid.size();
+    int width = grid[0].size();
+    std::vector<std::vector<uint8_t>> explored(height, std::vector<uint8_t>(width));
+    int max_island_area = 0;
+    for (int col = 0; col < height; ++col) {
+      for (int row = 0; row < width; ++row) {
+        if (grid[col][row] == WATER || explored[col][row]) {
+          continue;
+        }
+        max_island_area = std::max(max_island_area, CalculateIslandArea(col, row, grid, explored));
+      }
+    }
+    return max_island_area;
+  }
+
+private:
+  const int WATER = 0;
```

Comment:
定数であれば下記のようにすると良さそうです。
```suggestion
static constexpr int WATER = 0;
```

Reply:
ありがとうございます。

背景としては、`constexpr` にすると「コンパイル時に値が決定する」ことに意味があるように読めてしまうため、単に「定数である」ことを示す `const` にしました。

コンパイル時に値が決定することを特筆すべきでない場合でも、`static constexpr` を使った方が良いのでしょうか。